aws/resource_aws_ecr_replication_configuration_test.go

```diff
@@ -0,0 +1,99 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccAWSEcrReplicationConfiguration_basic(t *testing.T) {
+	resourceName := "aws_ecr_replication_configuration.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		ErrorCheck:   testAccErrorCheck(t, ecr.EndpointsID),
+		Providers:    testAccProviders,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSEcrReplicationConfiguration(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSEcrReplicationConfigurationExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "replication_configuration.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "replication_configuration.0.rule.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "replication_configuration.0.rule.0.destination.#", "1"),
+					resource.TestCheckResourceAttrSet(resourceName, "replication_configuration.0.rule.0.destination.0.region"),
```

Comment:
It might be worthwhile to use an explicit region, such as `testAccGetAlternateRegion()` so that we can test for a specific result, here and in the third step

Reply:
Chnaged, i hope that this is what you meant :D